src/table.rs

```diff
@@ -58,35 +63,134 @@ pub struct TableInner {
-    opt: Options,
+    opts: Options,
 }
 
 pub struct Table {
     inner: Arc<TableInner>,
 }
 
+/// `AsRef<TableInner>` is only used in `init_biggest_and_smallest`
+/// to construct a table iterator from `&TableInner`.
+impl AsRef<TableInner> for TableInner {
+    fn as_ref(&self) -> &TableInner {
+        self
+    }
+}
+
 impl TableInner {
     /// Create an SST from bytes data generated with table builder
-    pub fn create(_path: &Path, _data: Bytes, _opt: Options) -> Result<TableInner> {
-        unimplemented!()
+    fn create(path: &Path, data: Bytes, opts: Options) -> Result<TableInner> {
+        let mut f = fs::OpenOptions::new()
+            .create_new(true)
+            .read(true)
+            .write(true)
+            .open(path)?;
+        f.write(&data)?;
+        // TODO: pass file object directly to open and sync write
+        drop(f);
+        Self::open(path, opts)
     }
 
     /// Open an existing SST on disk
-    pub fn open(_path: &Path, _opt: Options) -> Result<TableInner> {
-        unimplemented!()
+    fn open(path: &Path, opts: Options) -> Result<TableInner> {
+        let f = fs::OpenOptions::new()
+            .read(true)
+            .write(false)
+            .create(false)
+            .open(path)?;
+        let file_name = path.file_name().unwrap().to_str().unwrap();
+        let id = parse_file_id(file_name)?;
+        let meta = f.metadata()?;
+        let table_size = meta.len();
+        let mut inner = TableInner {
+            file: MmapFile::File {
+                file: Mutex::new(f),
+                name: path.to_path_buf(),
+            },
+            table_size: table_size as usize,
+            smallest: Bytes::new(),
+            biggest: Bytes::new(),
+            id,
+            checksum: Bytes::new(),
+            estimated_size: 0,
+            index: TableIndex::default(),
+            index_start: 0,
+            index_len: 0,
+            opts,
+        };
+        inner.init_biggest_and_smallest()?;
+        // TODO: verify checksum
+        Ok(inner)
     }
 
     /// Open an existing SST from data in memory
-    pub fn open_in_memory(_data: Bytes, _id: u64, _opt: Options) -> Result<TableInner> {
-        unimplemented!()
+    fn open_in_memory(data: Bytes, id: u64, opts: Options) -> Result<TableInner> {
+        let table_size = data.len();
+        let mut inner = TableInner {
+            file: MmapFile::Memory { data },
+            opts,
+            table_size,
+            id,
+            smallest: Bytes::new(),
+            biggest: Bytes::new(),
+            checksum: Bytes::new(),
+            estimated_size: 0,
+            index: TableIndex::default(),
+            index_start: 0,
+            index_len: 0,
+        };
+        inner.init_biggest_and_smallest()?;
+        Ok(inner)
     }
 
     fn init_biggest_and_smallest(&mut self) -> Result<()> {
-        unimplemented!()
+        let ko = self.init_index()?;
+        self.smallest = Bytes::from(ko.key.clone());
```

Comment:
I thought it uses `Bytes` already, turned out I was wrong.

Reply:
Yes, we have modified this in previous PR